sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -314,16 +384,23 @@ public String generateSas(TableSasSignatureValues tableSasSignatureValues) {
     }
 
     /**
-     * Inserts an entity into the table if it does not exist, or merges the entity with the existing entity otherwise.
+     * Inserts an {@link TableEntity entity} into the table if it does not exist, or merges the
+     * {@link TableEntity entity} with the existing {@link TableEntity entity} otherwise.
+     *
+     * If no {@link TableEntity entity} exists within the table having the same partition key and row key as the
+     * provided {@link TableEntity entity}, it will be inserted. Otherwise, the provided {@link TableEntity entity}'s
+     * properties will be merged into the existing {@link TableEntity entity}.
      *
-     * If no entity exists within the table having the same partition key and row key as the provided entity, it will be
-     * inserted. Otherwise, the provided entity's properties will be merged into the existing entity.
+     * <p><strong>Code Samples</strong></p>
+     * <p>Upserts an {@link TableEntity entity} into the table. Prints out the details of the upserted
+     * {@link TableEntity entity}.</p>
+     * {@codesnippet com.azure.data.tables.tableAsyncClient.upsertEntity#TableEntity}
      *
-     * @param entity The entity to upsert.
+     * @param entity The {@link TableEntity entity} to upsert.
      *
-     * @return An empty reactive result.
+     * @return An empty {@link Mono}.
      *
-     * @throws IllegalArgumentException If the provided entity is invalid.
+     * @throws IllegalArgumentException If the provided {@link TableEntity entity} is invalid.
```

Comment:
What makes the `TableEntity` invalid?

Reply:
It being `null`, for example.

Reply:
Actually, that's the only case where we thorw. I will change the JavaDoc to reflect this.